hoot-services/src/main/java/hoot/services/controllers/osm/map/MapResource.java

```diff
@@ -443,7 +442,7 @@ public Response getTileNodesCounts(@Context HttpServletRequest request, String p
         long nodeCnt = 0;
         for (Object aParamsArray : paramsArray) {
             JSONObject param = (JSONObject) aParamsArray;
-            mapId = (String) param.get("mapId");
+            mapId = param.get("mapId").toString();
```

Comment:
So I know see the difference that necessitated this.  Hoot UI v1 sends a string for mapId:
```
[
  {
    "tile": "11.211547851562512,8.619041018922134,11.217041015625014,8.624472107633924",
    "mapId": "2"
  },
.
.
.
]
```

while Hoot UI v2x sends a number:
```
[
  {
    "id": "34812,31194,16_2",
    "xyz": [
      34812,
      31194,
      16
    ],
    "extent": {
      "0": [
        11.22802734375001,
        8.608178607442506
      ],
      "1": [
        11.233520507812509,
        8.613609852175768
      ]
    },
    "mapId": 2,
    "layerName": "ReviewPoisA",
    "tile": "11.22802734375001,8.608178607442506,11.233520507812509,8.613609852175768"
  },
.
.
.
]
```


I had not looked at the payloads before and I incorrectly assumed that the nodescount request was being made per tile just like the map data requests.  Instead I see that we send all the tile definitions for the view extent in the payload, but surprisingly in the dev tools network tab, I see numerous nodescount requests that do appear to be per tile.

So it needs to be either a single nodescount request with all tile definitions per view extent change, or we can model the nodescount request more like a map data request and do one per tile.  It could then be a GET that models the map request:
>http://localhost:8888/hoot-services/osm/api/0.6/nodescount/2/11.217041015625007,8.619041018922147,11.222534179687507,8.624472107633949

I think the kind of prefer the latter and don't think it would be that much less efficient (since the current code is making multiple identical requests already, i.e. Hoot UI v1 is making the same mistake in it's implementation which is why we probably inherited it here).

Reply:
Okay, I see now that it's not actually calling nodescount per tile, but it is calling it more time than it seems is necessary.